src/main/java/bio/terra/common/CloudUtil.java

```diff
@@ -0,0 +1,64 @@
+package bio.terra.common;
+
+import bio.terra.model.CloudPlatform;
+
+import java.util.Objects;
+import java.util.function.Supplier;
+
+/**
+ * Utility methods for working across cloud providers
+ */
+public class CloudUtil {
```

Comment:
To me, this doesn't solve the main problem, which is having to put different code for different platforms throughout our code base.

Instead of making it easier to `switch`, we should have an interface for different cloud operations and use subclasses to implement the operations for different platforms.

Reply:
Oh, I think I see.  I have to say in that case I don't agree.  We're switching based on values coming in...I don't see how we avoid having to to figure out the action.  We could add it to a subclass but I don't think that buys you much short have needing to go through more classes to understand the code.  You need to end up needing to instantiate these subclasses using a switch anyways...so I'm not totally sure what it buys you

Reply:
Ah, I see from your other PR...ok.  That makes more sense.  My concern there is in the size that this class might get.  Seems like having a single class do the branching is going to get massive pretty quickly.  It's also worth noting that we already have some abstraction at the individual class types (e.g. CloudStorage, Region, etc.).  That being said, I do like the idea of having some common methods like that in a common interface and having some way to instantiate it based on common objects...

Reply:
Cool, so I ended up incorporating your suggested implementation (couple super small changes around branching in the flights) but the last commit in the PR should have the thing